### PR TITLE
Pre-existing projects have a `description`

### LINKED ISSUE #197: Pre-existing projects don't have a `description`
In c770ff9ecc700b5f3d3876a87bb40952d4abadbb, @aminemboulouma introduced a project `description` field, and updated the `projects.html` template to display project descriptions on the website.

The problem is that, currently, in our database, we have `project` entries that were created before the `description` field existed, so they don't have that field (in our `db.json` file):

```js
    "firefox": {
      "id": "firefox",
      "name": "Firefox (git)",
      "icon": "/img/firefox.svg",
      [...]
    }
```

This will cause a bug when we try to generate the `projects.html` page. To fix that bug, we should add a "temporary migration code" to the [getProject](https://github.com/JanitorTechnology/janitor/blob/c770ff9ecc700b5f3d3876a87bb40952d4abadbb/lib/machines.js#L478) function that will check whether a `description` field is present in older projects, and add one if there is none.

Similar migration codes exist in the [lib/users.js](https://github.com/JanitorTechnology/janitor/blob/c770ff9ecc700b5f3d3876a87bb40952d4abadbb/lib/users.js#L347-L350) file.

### PR DESCRIPTION
**Pull request by aminemboulouma**
Fixed #197

### CODE REVIEW

File: lib/machines.js [hunk 1]
```diff
@@ -476,6 +476,9 @@ function getProject (projectId, create) {
     return null;
   }
 
+  if(!project.description)  {
```
Comment (jankeromnes) [inline lib/machines.js]: Nit: Please explain this code by adding a comment above this line, e.g. `// Temporary migration code: Previous projects didn't have a description.`.

Comment (jankeromnes) [reply lib/machines.js]: Nit: Our coding style normally has a space between `if` and `(`. Please add one.

File: lib/machines.js [hunk 2]
```diff
@@ -476,6 +476,9 @@ function getProject (projectId, create) {
...
+    project.description='';
```
Comment (jankeromnes) [inline lib/machines.js]: Nit: Our coding style normally has spaces before and after `=`, please add them.

File: lib/machines.js [hunk 3]
```diff
@@ -476,6 +476,9 @@ function getProject (projectId, create) {
...
+  }
```
Comment (jankeromnes) [inline lib/machines.js]: Nit: Please add an empty line below this `}` in order to separate the two code blocks.

**Comment by aminemboulouma** [comment]
Great to know ! I have updated my code with all the nits and added the comment.
Great to know how you are working on the project, thank you for sharing  that with me !

File: lib/machines.js [hunk 4]
```diff
@@ -476,6 +476,11 @@ function getProject (projectId, create) {
     return null;
   }
 
+// Temporary migration code: Previous projects didn't have a description.
```
Comment (jankeromnes) [inline lib/machines.js]: Nit: Please indent this line with 2 spaces.

File: lib/machines.js [hunk 5]
```diff
@@ -476,6 +476,11 @@ function getProject (projectId, create) {
...
+  if (!project.description)  {
```
Comment (jankeromnes) [inline lib/machines.js]: Nit: Please remove one of the two spaces before `{`.

**Comment by aminemboulouma** [comment]
I think it is okay now, what do you think about it ?

**Comment by jankeromnes** [comment]
> I think it is okay now, what do you think about it ?

Thanks! You fixed the double space issue. However Travis still found two more issues:

> /home/travis/build/JanitorTechnology/janitor/lib/machines.js
>   479:1  error  Expected indentation of 2 spaces but found 0  indent
>   483:1  error  Trailing spaces not allowed                   no-trailing-spaces

Note that you can simply run `npm run lint-fix` in the IDE to automatically fix all these issues. 😄 (Sorry, I should have mentioned this command earlier!)

**Comment by aminemboulouma** [comment]
It is okay, I have done so, is there anything I forgot about ?

**Comment by jankeromnes** [comment]
All green! Ready for launch 🚀✔️

**Comment by jankeromnes** [comment]
Merged! 🏁 Thanks again!

**Pull request merged**